### PR TITLE
USHIFT-261: clean up path handling for kubeconfig files

### PR DESCRIPTION
**Pull request by dhellmann**
Provide a consistent way to calculate the path to a set of managed
kubeconfig files so we can avoid errors with path management
throughout the code.

### CODE REVIEW

File: pkg/config/config.go [hunk 1]
```diff
@@ -78,6 +78,21 @@ type MicroshiftConfig struct {
...
+
+const (
+	KubeAdmin             KubeConfigID = "kubeadmin"
+	KubeControllerManager KubeConfigID = "kube-controller-manager"
+	KubeScheduler         KubeConfigID = "kube-scheduler"
+	Kubelet               KubeConfigID = "kubelet"
+)
+
+// KubeConfig returns the path to the specified kubeconfig file.
+func (cfg *MicroshiftConfig) KubeConfig(id KubeConfigID) string {
```
Comment (stlaz) [inline pkg/config/config.go:92]: Let's just call it `KubeConfigPath()`

File: pkg/components/components.go [hunk 2]
```diff
@@ -1,33 +1,33 @@
...
 import (
-	"path/filepath"
-
 	"github.com/openshift/microshift/pkg/config"
 	"k8s.io/klog/v2"
 )
 
 func StartComponents(cfg *config.MicroshiftConfig) error {
-	if err := startServiceCAController(cfg, filepath.Join(cfg.DataDir, "/resources/kubeadmin/kubeconfig")); err != nil {
+	kubeAdminConfig := cfg.KubeConfig(config.KubeAdmin)
```
Comment (stlaz) [inline pkg/components/components.go:9]: :+1:

**Review by stlaz — COMMENTED** [review]
looks good, I think I would be just a little bit more explicit with the name of the function that returns the path

File: pkg/config/config.go (hunk 1 above)
Comment (dhellmann) [reply pkg/config/config.go]: Good call, done

**Pull request merged**